### ISSUE TITLE
Crash after a shortcut

### ISSUE THREAD

**Issue by lolo32** [curlpipe/ox#83]
**Describe the bug**
The editor crash on shortcut pressed.

**To Reproduce**
1. Launch with `ox`
2. I use the shortcut `Ctrl+?` but on my french keyboard it's written `Ctrl+Shift+?`, two time
3. the editor crashed

**Desktop (please complete the following information):**
 - OS: Debian
 - Version: 0.2.6

**Backtrace:**
```
thread 'main' panicked at 'index out of bounds: the len is 0 but the index is 0', src/document.rs:385:21
    stack backtrace:
    0:     0x55fc40811293 - std::backtrace_rs::backtrace::libunwind::trace::h577ea05e9ca4629a
                at /rustc/18bf6b4f01a6feaf7259ba7cdae58031af1b7b39/library/std/src/../../backtrace/src/backtrace/libunwind.rs:96
    1:     0x55fc40811293 - std::backtrace_rs::backtrace::trace_unsynchronized::h50b9b72b84c7dd56
                at /rustc/18bf6b4f01a6feaf7259ba7cdae58031af1b7b39/library/std/src/../../backtrace/src/backtrace/mod.rs:66
    2:     0x55fc40811293 - std::sys_common::backtrace::_print_fmt::h6541cf9823837fac
                at /rustc/18bf6b4f01a6feaf7259ba7cdae58031af1b7b39/library/std/src/sys_common/backtrace.rs:79
    3:     0x55fc40811293 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::hf64fbff071026df5
                at /rustc/18bf6b4f01a6feaf7259ba7cdae58031af1b7b39/library/std/src/sys_common/backtrace.rs:58
    4:     0x55fc4079451c - core::fmt::write::h9ddafa4860d8adff
                at /rustc/18bf6b4f01a6feaf7259ba7cdae58031af1b7b39/library/core/src/fmt/mod.rs:1082
    5:     0x55fc408108f6 - std::io::Write::write_fmt::h1d2ee292d2b65481
                at /rustc/18bf6b4f01a6feaf7259ba7cdae58031af1b7b39/library/std/src/io/mod.rs:1514
    6:     0x55fc408102ea - std::sys_common::backtrace::_print::ha25f9ff5080d886d
                 at /rustc/18bf6b4f01a6feaf7259ba7cdae58031af1b7b39/library/std/src/sys_common/backtrace.rs:61
    7:     0x55fc408102ea - std::sys_common::backtrace::print::h213e8aa8dc5405c0
                 at /rustc/18bf6b4f01a6feaf7259ba7cdae58031af1b7b39/library/std/src/sys_common/backtrace.rs:48
    8:     0x55fc408102ea - std::panicking::default_hook::{{closure}}::h6482fae49ef9d963
                 at /rustc/18bf6b4f01a6feaf7259ba7cdae58031af1b7b39/library/std/src/panicking.rs:200
    9:     0x55fc4080fbb7 - std::panicking::default_hook::he30ad7589e0970f9
                at /rustc/18bf6b4f01a6feaf7259ba7cdae58031af1b7b39/library/std/src/panicking.rs:219
   10:     0x55fc4080fbb7 - std::panicking::rust_panic_with_hook::haa1ed36ada4ffb03
                at /rustc/18bf6b4f01a6feaf7259ba7cdae58031af1b7b39/library/std/src/panicking.rs:569
   11:     0x55fc4080f84a - std::panicking::begin_panic_handler::{{closure}}::h7001af1bb21aeaeb
                at /rustc/18bf6b4f01a6feaf7259ba7cdae58031af1b7b39/library/std/src/panicking.rs:476
   12:     0x55fc4080f814 - std::sys_common::backtrace::__rust_end_short_backtrace::h39910f557f5f2367
                 at /rustc/18bf6b4f01a6feaf7259ba7cdae58031af1b7b39/library/std/src/sys_common/backtrace.rs:153
   13:     0x55fc4080f7cd - rust_begin_unwind
                at /rustc/18bf6b4f01a6feaf7259ba7cdae58031af1b7b39/library/std/src/panicking.rs:475
   14:     0x55fc40791ce0 - core::panicking::panic_fmt::h4e2659771ebc78eb
                at /rustc/18bf6b4f01a6feaf7259ba7cdae58031af1b7b39/library/core/src/panicking.rs:85
   15:     0x55fc40791b71 - core::panicking::panic_bounds_check::h2e8c50d2fb4877c0
                at /rustc/18bf6b4f01a6feaf7259ba7cdae58031af1b7b39/library/core/src/panicking.rs:62
   16:     0x55fc4072311c - ox::editor::Editor::execute::h41a91b916b54259e
   17:     0x55fc407346e4 - ox::main::he0e8202042f9c811
   18:     0x55fc406f5043 - std::sys_common::backtrace::__rust_begin_short_backtrace::hcf14f977209b4ed6
   19:     0x55fc4073ecb5 - main
   20:     0x7f69db61209b - __libc_start_main
   21:     0x55fc406f307a - _start
   22:                0x0 - <unknown>
 Aborted
```

**Comment by curlpipe** [comment]
What does the keyboard shortcut `Ctrl+?` do?

**Comment by lolo32** [comment]
Nothing special, I tried it if it could display a window with all keyboard shortcut (some sort of help).

It's just a fresh installation, with nothing more nor configuration

**Comment by curlpipe** [comment]
I can't seem to reproduce this issue. Try typing some text first and then try pressing the shortcut. Does it delete the text?

**Comment by lolo32** [comment]
ok, I reproduce it pressing two time backspace (this seems to be the same as `Ctrl+Shift+?` and when my document is empty, it will crash.
It crash every times I push 2 times backspace at the beginning of the document, event if there is some text later in the file.

**Comment by curlpipe** [comment]
This has already been fixed in 0.2.7 which is still being worked on. I hope to release it soon. :)
